plugin/src/main/java/org/opensearch/ml/model/MLModelGroupManager.java

```diff
@@ -0,0 +1,195 @@
+                listener.onFailure(e);
+            }
+        } catch (final Exception e) {
+            log.error("Failed to init model group index", e);
+            listener.onFailure(e);
+        }
+    }
+
+    private void validateRequestForAccessControl(MLRegisterModelGroupInput input, User user) {
+        AccessMode modelAccessMode = input.getModelAccessMode();
+        Boolean isAddAllBackendRoles = input.getIsAddAllBackendRoles();
+        if (modelAccessMode == null) {
+            if (modelAccessMode == null) {
+                if (!CollectionUtils.isEmpty(input.getBackendRoles()) && Boolean.TRUE.equals(isAddAllBackendRoles)) {
+                    throw new IllegalArgumentException("You cannot specify backend roles and add all backend roles at the same time.");
+                } else if (Boolean.TRUE.equals(isAddAllBackendRoles) || !CollectionUtils.isEmpty(input.getBackendRoles())) {
+                    input.setModelAccessMode(AccessMode.RESTRICTED);
+                    modelAccessMode = AccessMode.RESTRICTED;
+                } else {
+                    input.setModelAccessMode(AccessMode.PRIVATE);
+                }
+            }
+        }
+        if ((AccessMode.PUBLIC == modelAccessMode || AccessMode.PRIVATE == modelAccessMode)
+            && (!CollectionUtils.isEmpty(input.getBackendRoles()) || Boolean.TRUE.equals(isAddAllBackendRoles))) {
+            throw new IllegalArgumentException("You can specify backend roles only for a model group with the restricted access mode.");
+        } else if (AccessMode.RESTRICTED == modelAccessMode) {
+            if (modelAccessControlHelper.isAdmin(user) && Boolean.TRUE.equals(isAddAllBackendRoles)) {
+                throw new IllegalArgumentException("Admin users cannot add all backend roles to a model group.");
+            }
+            if (CollectionUtils.isEmpty(user.getBackendRoles())) {
+                throw new IllegalArgumentException("You must have at least one backend role to register a restricted model group.");
+            }
+            if (CollectionUtils.isEmpty(input.getBackendRoles()) && !Boolean.TRUE.equals(isAddAllBackendRoles)) {
+                throw new IllegalArgumentException(
+                    "You must specify one or more backend roles or add all backend roles to register a restricted model group."
+                );
+            }
+            if (!CollectionUtils.isEmpty(input.getBackendRoles()) && Boolean.TRUE.equals(isAddAllBackendRoles)) {
+                throw new IllegalArgumentException("You cannot specify backend roles and add all backend roles at the same time.");
+            }
+            if (!modelAccessControlHelper.isAdmin(user)
+                && !Boolean.TRUE.equals(isAddAllBackendRoles)
+                && !new HashSet<>(user.getBackendRoles()).containsAll(input.getBackendRoles())) {
+                throw new IllegalArgumentException("You don't have the backend roles specified.");
+            }
+        }
+    }
```

Comment:
Can we add comments for all these use cases?

Reply:
Yeah will add comments